ld/testsuite/
2012-07-26  Meador Inge  <meadori@codesourcery.com>

	PR ld/14397
	* ld-arm/arm-elf.exp: Add tests of linking without .rel.plt.
	* ld-arm/arm-no-rel-plt.out: Linker script without .rel.plt section.
	* ld-arm/arm-no-rel-plt.out: Expected output.

bfd/
2012-07-26  Meador Inge  <meadori@codesourcery.com>
	    Nick Clifton  <nickc@redhat.com>

	PR ld/14397
	* elf32-arm.c (elf32_arm_finish_dynamic_sections): Report an error
	if a required section is missing from the linker script.

===================================================================
RCS file: /cvs/src/src/bfd/elf32-arm.c,v
retrieving revision 1.297
retrieving revision 1.298
diff -u -r1.297 -r1.298
--- src/bfd/elf32-arm.c	2012/07/24 21:06:57	1.297
+++ src/bfd/elf32-arm.c	2012/07/26 13:43:23	1.298
@@ -14083,7 +14083,14 @@
 	      name = RELOC_SECTION (htab, ".plt");
 	    get_vma:
 	      s = bfd_get_section_by_name (output_bfd, name);
-	      BFD_ASSERT (s != NULL);
+	      if (s == NULL)
+		{
+		  /* PR ld/14397: Issue an error message if a required section is missing.  */
+		  (*_bfd_error_handler)
+		    (_("error: required section '%s' not found in the linker script"), name);
+		  bfd_set_error (bfd_error_invalid_operation);
+		  return FALSE;
+		}
 	      if (!htab->symbian_p)
 		dyn.d_un.d_ptr = s->vma;
 	      else
